include/CBAWFile.h

```diff
@@ -179,63 +209,66 @@ namespace core
 	} PACK_STRUCT;
 
 	//! Simple struct of essential data of ICPUMeshBuffer that has to be exported
-	struct MeshBufferBlobV0 : TypedBlob<MeshBufferBlobV0, scene::ICPUMeshBuffer>
+	struct MeshBufferBlobV0 : TypedBlob<MeshBufferBlobV0, scene::ICPUMeshBuffer>, FixedSizeBlob<MeshBufferBlobV0, scene::ICPUMeshBuffer>
 	{
 		//! Constructor filling all members
 		explicit MeshBufferBlobV0(const scene::ICPUMeshBuffer*);
 
 		video::SMaterial mat;
 		core::aabbox3df box;
 		uint64_t descPtr;
-		video::E_INDEX_TYPE indexType;
+		uint32_t indexType;
 		uint32_t baseVertex;
 		uint64_t indexCount;
 		size_t indexBufOffset;
 		size_t instanceCount;
 		uint32_t baseInstance;
-		scene::E_PRIMITIVE_TYPE primitiveType;
-		scene::E_VERTEX_ATTRIBUTE_ID posAttrId;
+		uint32_t primitiveType;
+		uint32_t posAttrId;
 	} PACK_STRUCT;
 
-	struct SkinnedMeshBufferBlobV0 : TypedBlob<SkinnedMeshBufferBlobV0, scene::SCPUSkinMeshBuffer>
+	struct SkinnedMeshBufferBlobV0 : TypedBlob<SkinnedMeshBufferBlobV0, scene::SCPUSkinMeshBuffer>, FixedSizeBlob<SkinnedMeshBufferBlobV0, scene::SCPUSkinMeshBuffer>
 	{
 		//! Constructor filling all members
 		explicit SkinnedMeshBufferBlobV0(const scene::SCPUSkinMeshBuffer*);
 
 		video::SMaterial mat;
 		core::aabbox3df box;
 		uint64_t descPtr;
-		video::E_INDEX_TYPE indexType;
+		uint32_t indexType;
 		uint32_t baseVertex;
 		uint64_t indexCount;
 		size_t indexBufOffset;
 		size_t instanceCount;
 		uint32_t baseInstance;
-		scene::E_PRIMITIVE_TYPE primitiveType;
-		scene::E_VERTEX_ATTRIBUTE_ID posAttrId;
+		uint32_t primitiveType;
+		uint32_t posAttrId;
 		uint32_t indexValMin;
 		uint32_t indexValMax;
 		uint32_t maxVertexBoneInfluences;
 	} PACK_STRUCT;
 
 	//! Simple struct of essential data of ICPUMeshDataFormatDesc that has to be exported
-	struct MeshDataFormatDescBlobV0 : TypedBlob<MeshDataFormatDescBlobV0, scene::IMeshDataFormatDesc<core::ICPUBuffer> >
+	struct MeshDataFormatDescBlobV0 : TypedBlob<MeshDataFormatDescBlobV0, scene::IMeshDataFormatDesc<core::ICPUBuffer> >, FixedSizeBlob<MeshDataFormatDescBlobV0, scene::IMeshDataFormatDesc<core::ICPUBuffer> >
 	{
+	private:
+		enum { VERTEX_ATTRIB_CNT = 16 };
```

Comment:
can we make it equal to scene::EVAI_COUNT ?

or do we run into circular refs?

Reply:
> or do we run into circular refs?  
  
Yes, I had to forward declare every class used by blobs to make them able to inherit from classes from CBAWFile.h. So there's the best solution I came up with. I put assert (or rather `_IRR_DEBUG_BREAK_IF` so we will know 
when `EVAI_COUNT` change https://github.com/buildaworldnet/IrrlichtBAW/pull/36/commits/22ddd15def5d4041b3480288da97e5c6d7859784#diff-b7b624bc03dce20e1e4c837d54b34971R103

Reply:
good solution